src/main/scala/viper/gobra/frontend/info/implementation/typing/ghost/separation/GhostTyping.scala

```diff
@@ -154,7 +154,12 @@ trait GhostTyping extends GhostClassifier { this: TypeInfoImpl =>
         case x => noGhostPropagationFromChildren(x)
       }
 
-      tree.child(node).forall(noGhostPropagationFromSelfAndChildren)
+      val childrenToVisit: Iterable[PNode] = node match {
+        case PUnfolding(_, op) => Vector(op)
+        case _ => tree.child(node)
+      }
```

Comment:
I do not understand this fix. PUnfolding should only be ghost if the children are ghost as implemented in the ghostExprClassification (ghostExpryping).

Reply:
I meant the argument expression, not all children.

Reply:
```
// ghostness of proof annotations is decided by the argument
      case ann: PActualExprProofAnnotation => ghost(!noGhostPropagationFromChildren(ann.op))
```